### ISSUE TITLE
post-comment with ID

### ISSUE THREAD

**Issue by zimbatm** [zimbatm/actions-cli#7]
https://twitter.com/timbertson/status/1245081280657514496?s=20
> Cool! I'd love a variant which did add-or-update comment, so it doesn't add a new comment on every push to a long PR. Maybe using a cache key to store existing comment ID?

---

I don't see many locations aside from the comment itself to store the cache key. Maybe something like this?

    action-cli post-comment --comment-id "my-id" "message"

```
<!-- my-id -->
message
```

**Comment by rizary** [comment]
Maybe we can edit comment that already being posted? Like every time someone push into the existing PR, the comment will not be added twice but instead revise current comment.

**Comment by zimbatm** [comment]
yes that's the idea

**Comment by timbertson** [comment]
I was thinking you'd post a comment, then store the comment ID (as in, the github generated ID) with however `actions/cache` stores stuff. But the predetermined ID in a comment is much nicer, because failures don't affect it.

**Comment by zimbatm** [comment]
One thing to verify is if the default GITHUB_TOKEN has the permissions to read and update the comments

**Comment by DrSensor** [comment]
Or maybe store the id as [*git-notes*](https://git-scm.com/docs/git-notes) 🤔

References:
* [Gerrit `refs/notes/review`](https://gerrit.googlesource.com/plugins/reviewnotes/%2B/master/src/main/resources/Documentation/refs-notes-review.md)
* [Git Tip of the Week: Git Notes](https://alblue.bandlem.com/2011/11/git-tip-of-week-git-notes.html)
* [more uses for git notes, and hidden treasures in Gerrit](https://blog.adamspiers.org/2013/10/02/more-uses-for-git-notes/)
* [Git Dark Magic](https://gist.github.com/DrSensor/d7d005265009b38f2130adc27eae59be#file-git-notes-md) (just my old notes about *git-notes*)
* [my attempt storing perf data on GitHub as *git-notes*](https://github.com/DrSensor/scdlang/commits/refs/notes/commits)

**Comment by zimbatm** [comment]
Are they being displayed in the PR view? The main motivation was to allow posting formatting and linting issues on PRs.

**Comment by DrSensor** [comment]
No. It's hidden in the git ref object and can only be accessed via specific urls or explicitly fetch them via `git fetch $remote refs/notes/$name:refs/notes/$name`.

`git clone` will *not* automatically fetch the notes. So the .git folder will stay as it was (unless you fetch them) and the git history will stay clean (whatever you fetch them or not).

Currently, no GitHub UI or button to easily access it, just url.
